client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -2193,26 +2192,20 @@ Undocumented.prototype.getExport = function( siteId, exportId, fn ) {
 /**
  * Check different info about WordPress and Jetpack status on a url
  *
- * @param {String} targetUrl - The url of the site to check
- * @returns {Promise}  promise
+ * @param  {string}  inputUrl The url of the site to check. Must use http or https protocol.
+ * @return {Promise} promise  Request promise
  */
-Undocumented.prototype.getSiteConnectInfo = function( targetUrl ) {
-	const parsedUrl = url.parse( targetUrl );
-	let endpointUrl = `/connect/site-info/${ parsedUrl.protocol.slice( 0, -1 ) }/${ parsedUrl.host }`;
-
-	if ( parsedUrl.path && parsedUrl.path !== '/' ) {
-		endpointUrl += parsedUrl.path.replace( /\//g, '::' );
-	}
-
-	return this.wpcom.req.get( endpointUrl );
+Undocumented.prototype.getSiteConnectInfo = function( inputUrl ) {
+	return this.wpcom.req.get( '/connect/site-info', { url: inputUrl } );
```

Comment:
Now that the url moves to the query, we can rely on existing underlying mechanisms to handle escaping ⭐️